deps/wazuh_testing/wazuh_testing/process_resource_monitoring/src/wazuh_metrics.py

```diff
@@ -0,0 +1,334 @@
+    )
+    parser.add_argument(
+        '--store-disk',
+        dest='store_disk_path',
+        action='store',
+        default=f'{CURRENT_SESSION}/files',
+        type=str,
+        help=f'Path to store the CSVs with the disk usage data. Default: \'{METRICS_FOLDER}/<id>/files\'.\n'
+        + 'Where <id> is the number of seconds since the epoch when the monitoring started.',
+    )
+
+    return parser.parse_args()
+
+
+def shutdown_threads(signal_number: int, frame: Optional[FrameType]) -> None:
+    """Stop and close all monitoring threads.
+
+    Args:
+        signal_number (int): number of the signal.
+        frame (FrameType, optional): frame of the object.
+    """
+    logger.info('Attempting to shutdown all monitor threads')
+
+    global session_active
+    session_active = False
+
+    # Shutdown all possible monitors
+    for monitor in sum(active_monitors.values(), []):
+        monitor.shutdown()
+
+    # Shutdown all disk trackers
+    for tracker in active_disk_trackers.values():
+        tracker.shutdown()
+
+    logger.info('Process finished gracefully')
+
+
+def check_monitors_health(options) -> bool:
+    """Write the collected data in a CSV file.
+
+    Args:
+        options (argparse.Options): object containing the script options.
+
+    Returns:
+        (bool): False if there were any errors. True otherwise.
+    """
+    healthy = True
+    for process, monitors in active_monitors.items():
+        # Check if there is any unhealthy monitor
+        if any(filter(lambda monitor: monitor.is_event_set(), monitors)):
+            logger.warning(f'Monitoring of {process} failed. Attempting to create new monitor instances')
+
+            try:
+                # Try to get new PIDs
+                process_pids = Monitor.get_process_pids(process)
+                # Shutdown all the related monitors to the failed process (necessary for multiprocessing)
+                for monitor in monitors:
+                    monitor.shutdown()
+            except ValueError:
+                healthy = False
+                logger.warning(f'Could not create new monitor instances for {process}')
+                continue
+
+            for i, pid in enumerate(process_pids):
+                # Attempt to create new monitor instances for the process
+                p_name = process if i == 0 else f'{process}_child_{i}'
+                monitor = Monitor(
+                    process_name=p_name,
+                    pid=pid,
+                    value_unit=options.data_unit,
+                    time_step=options.sleep_time,
+                    version=options.version,
+                    dst_dir=options.store_process_path,
+                )
+                monitor.start()
+
+                try:
+                    # Replace old monitors for new ones
+                    active_monitors[process][i] = monitor
+                except IndexError:
+                    active_monitors[process].append(monitor)
```

Comment:
In the case of the dynamic amount of threads used, we will miss extra daemons launched in the middle of the scan. We need to handle that case

Reply:
Working on it, currently added https://github.com/wazuh/wazuh-qa/pull/5747/commits/067d45b0e0aae6724dbc28d326d44d908cba4494

Reply:
Finished it on https://github.com/wazuh/wazuh-qa/pull/5747/commits/c9572944b9012dbf636f2c00ce82012abb06edac